src/System.IO.Compression/tests/ZipArchive/zip_CreateTests.cs

```diff
@@ -149,6 +149,88 @@ public static void CreateUncompressedArchive()
             }
         }
 
+        [Theory]
+        [InlineData("minimalTwoFiles")]
+        public static void CreateNormal_VerifyDataDescriptor(string folder)
+        {
+            void CreateEntry(ZipArchive archive, string fileName, string fileContents)
+            {
+                ZipArchiveEntry entry = archive.CreateEntry(fileName);
+                using (StreamWriter writer = new StreamWriter(entry.Open()))
+                {
+                    writer.Write(fileContents);
+                }
+            }
+
+            string fileName = folder + ".zip";
+
+            if (File.Exists(fileName))
+            {
+                File.Delete(fileName);
+            }
+
+            using (var fileStream = new FileStream(fileName, FileMode.Create))
+            {
+                // We need an unseekable stream so the data descriptor bit is turned on when saving
+                var wrappedStream = new WrappedStream(fileStream, true, true, false, null);
+
+                // Creation will go through the path that sets the data descriptor bit when the stream is unseekable
+                using (var archive = new ZipArchive(wrappedStream, ZipArchiveMode.Create))
+                {
+                    CreateEntry(archive, "A", "xxx");
+                    CreateEntry(archive, "B", "yyy");
+                }
+            }
+
+            using (var fileStream = new FileStream(fileName, FileMode.Open))
```

Comment:
All this "check flags bytes" code is 13 lines and you're repeating it. consider extracting a method or local method. it's also more readable if you can write `AssertFlags(8)`

Reply:
I see what you mean. Thanks, yes, I'll put it in its own method.